_src/app/habitat/Habitat.js

```diff
@@ -208,12 +212,23 @@ define([
                 }
             }];
             this.initGrid(columns);
+
+            this.own(
+                on(this.grid, 'keyup', () => {
+                    this.showValidation(this.wettedWidthTxt.valueAsNumber);
+                }),
+                on(this.grid, 'dgrid-datachange', () => {
```

Comment:
dgrid-cellfocusin
dgrid-cellfocusout
dgrid-editor-show	
dgrid-editor-hide

do not fire with the current tabbing code. The double check is not a huge deal.

Reply:
Somethings not quite right still with the validation. Here are the steps to get to this state:
![image](https://user-images.githubusercontent.com/1326248/41103562-a7a5df58-6a26-11e8-8bc4-ca419e059da1.png)

1. Set the WW to 2
1. Click into SB cell and type 3
1. Click outside of the grid rather than hitting tab

It looks like to me that the validation code is still firing but it may be using old grid data?

Reply:
Hmm data change should fire in that instance? I'll try it and see.

Reply:
yup still has old store data.